code/modules/mob/interactive.dm

```diff
@@ -119,29 +119,34 @@
 		retal_target = potentialAssault
 	..()
 
-/mob/living/carbon/human/interactive/verb/customiseSNPC()
+/client/proc/customiseSNPC(var/mob/A in world)

```

Comment:
@Giacom for base types (Mob, Obj, Atom, Movable etc.) Byond makes an attempt at speeding it up.
it remains slower than a maintained list iirc.

but my real complaint here is it's a proc that only works on SNPCs, and so you have to search through about 500 mobs to find any that might be SNPCs (since SNPCs have normal human names this makes it also harder)

So yeah performance of it isn't great, but complaint more focused on usability of the verb.